### PR TITLE
feat: Cannon-compatible binary

### PR DESCRIPTION
**Pull request by ImTei**
**Description**

Implement Asterisc binary with Cannon-compatible commands:
- `load-elf`
- `witness`
- `run`

Almost every code is based on Cannon's implementation with minor fixes.

If the codes are related to VM or its other functionalities, I just copied and reused those codes from Cannon to minimize dependencies on Cannon and keep it as an alternative implementation. Otherwise if the codes are just utilities unrelated to VM, they are imported as dependencies. 

For the special case, flags of each command are directly imported from Cannon as a dependency because the interface of Asterisc binary must be compatible with Cannon for op-challenger integration.

This PR depends on modification on Optimism monorepo(https://github.com/ethereum-optimism/optimism/pull/9350). The dependency is temporarily [set to the working branch](https://github.com/ethereum-optimism/asterisc/pull/8/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R8) for now, and it will be changed to the develop branch once the optimism PR is merged.

**Tests**

Tested each command for `minimal.go`. More test cases could be added in future PRs, and this may be further tested and modified as we work on the op-challenger integration.

### CODE REVIEW

File: rvgo/cmd/log.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
```
Comment (ajsutton) [inline rvgo/cmd/log.go:8]: As a heads up, go-ethereum has just pulled in upstream geth changes which changes the logging library. https://github.com/ethereum-optimism/optimism/pull/8917 updated the monorepo for the new logging changes.

Doesn't need to be handled in this PR but probably is worth pulling in the updated op-geth soon.

File: rvgo/cmd/run.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+	l := Logger(os.Stderr, log.LvlInfo)
+	outLog := &LoggingWriter{Name: "program std-out", Log: l}
+	errLog := &LoggingWriter{Name: "program std-err", Log: l}
+
+	stopAtPreimageType := ctx.String(cannon.RunStopAtPreimageTypeFlag.Name)
+	if stopAtPreimageType != "" && stopAtPreimageType != "any" && stopAtPreimageType != "local" && stopAtPreimageType != "global" {
+		return fmt.Errorf("invalid preimage type %q, must be either 'any', 'local' or 'global'", stopAtPreimageType)
+	}
+	stopAtPreimageLargerThan := ctx.Int(cannon.RunStopAtPreimageLargerThanFlag.Name)
```
Comment (ajsutton) [inline rvgo/cmd/run.go:74]: These options are only used to help us test `op-challenger` so not strictly necessary in asterisc but for the record, they changed a bit in https://github.com/ethereum-optimism/optimism/pull/9406 so we can find the additional preimage types added for 4844 support.

**Review by ajsutton — APPROVED** [review]
LGTM. Just added a couple of notes about related PRs.

**Review by clabby — APPROVED** [review]
This looks good to me. Approving pending the monorepo changes being merged + updating the dependency :)

**Comment by ImTei** [comment]
@ajsutton Thank you for the updates about Canon changes. I have updated optimism dependency to the latest develop branch and applied log library changes and blob preimage key handling.

**Review by ajsutton — APPROVED** [review]
Nice. :)

**Pull request merged**